airflow/api_connexion/endpoints/task_instance_endpoint.py

```diff
@@ -326,7 +326,13 @@ def post_set_task_instances_state(*, dag_id: str, session: Session = NEW_SESSION
             detail=f"Task instance not found for task {task_id!r} on execution_date {execution_date}"
         )
 
-    if run_id and not session.query(TI).get({'task_id': task_id, 'dag_id': dag_id, 'run_id': run_id}):
+    if (
+        run_id
+        and session.query(TI)
+        .filter(TI.task_id == task_id, TI.dag_id == dag_id, TI.run_id == run_id)
+        .one_or_none()
+        is None
+    ):
```

Comment:
nit -- the following is more readable to me but no strong opinion 

```suggestion
    if run_id and (
        session.query(TI)
        .filter(TI.task_id == task_id, TI.dag_id == dag_id, TI.run_id == run_id)
        .one_or_none()
        is None
    ):
```